src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-param-reassign */
+
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books = [...state.books, {
+        title: action.payload.title,
+        author: action.payload.author,
+        genre: action.payload.genre,
+        percentage: 0,
+        chapter: 'Not opened yet',
+      }];
+    },
+    removeBook: (state, action) => {
+      const bookId = action.payload;
+      initialState.books = state.books.filter((book) => book.id !== bookId);
```

Comment:
- [ ] I think this line should be 

` state.books = state.books.filter((book) => book.id !== bookId);` 
In order to properly update the books array in the state object. The initialState object should not be modified in a reducer function.

Reply:
- [ ] It would be a good idea to add an ID when adding a new book on the reducer, something similar to:

`id: Math.floor(Math.random() * 100),`

Reply:
Thank you, @GzimAsani, I was testing the extent of the reducers and forgot to set `initialState` back to `state`. I will also add the ID, thank you for the advice. 👍

Reply:
I am glad I could help 🤗